lib/src/connector/connection.cc

```diff
@@ -173,6 +180,8 @@ void Connection::connect(int max_connect_attempts) {
             connectAndWait();
             if (connection_state_.load() == ConnectionStateValues::open)
                 return;
+            // Connection attempt failed, next try should be against a failover server.
```

Comment:
At this point, you know that `client_metadata_.connection_timeout` did expire; it's not a failure yet. We should not switch broker, as you'll continue iterating. Otherwise, with the first (failed) connection attempt, you'll end up switching twice.

In fact, with the broker down and logging "### Hey!" here, this is what I get when I start pxp-agent: https://gist.github.com/parisiale/3ffbdc99dde108ec73087f4aa33545b6